insights/specs/datasources/pcp.py

```diff
@@ -74,3 +91,38 @@ def pmlog_summary_args(broker):
         raise SkipComponent("Failed to get pmlogger metrics: {0}".format(str(e)))
 
     return "{0} {1}".format(pm_file, ' '.join(sorted(metrics)))
+
+
+@datasource(ros_collect, HostnameShort, HostContext)
+def pcp_raw_files(broker):
+    """
+    Return all existing paths of PCP RAW data files of yesterday.
+
+    .. note::
+        The "hostname" in the file paths will be removed by the "save_as"
+        in the `pcp_raw_data` spec.
+        All the collected PCP RAW data files will be stored in the
+        "var/log/pcp/pmlogger" directory in the archive.
+
+    Returns:
+        list(str): list of file path
+
+    Raises:
+        ContentException: Raises when number of PCP RAW files less than 3.
+    """
+    hostname = broker[HostnameShort].hostname
```

Comment:
On my system I can see directory has been created with hostname+domain and because of that we are failing to collect files,
~~~
ls /var/log/pcp/pmlogger/localhost.localdomain/
20240123.2.xz  20240130.index    20240131.index        20240201.13.27.meta

# hostname
localhost.localdomain
~~~

I can see there is another class which can be useful here,
~~~
(Pdb) broker[HostnameShort].hostname
'localhost'
(Pdb) from insights.parsers.hostname import HostnameDefault
(Pdb) broker[HostnameDefault].raw
'localhost.localdomain'
~~~

I think the best way to handle this is, we look for both full name and short name for yesterday's files. To go further here, I was checking how pcp gets the hostname and I can see its using libc gethostname https://man7.org/linux/man-pages/man2/gethostname.2.html . 
There it has mentioned that hostname can be truncated if its longer, I don't know how we can handle that case?!

Reply:
Sure, I changed to use the `Hostname` parser  and checks both the short hostname and fqdn